app/grandchallenge/archives/serializers.py

```diff
@@ -94,41 +89,15 @@ def __init__(self, *args, **kwargs):
 
     def create(self, validated_data):
         if validated_data.pop("values") != []:
-            raise ValidationError("Values can only be added via update")
+            raise DRFValidationError("Values can only be added via update")
         return super().create(validated_data)
 
     def update(self, instance, validated_data):
-        civs = validated_data.pop("values")
-
-        civ_pks_to_add = set()
-        upload_pks = {}
-
-        for civ in civs:
-            interface = civ.pop("interface", None)
-            upload_session = civ.pop("upload_session", None)
-            value = civ.pop("value", None)
-            image = civ.pop("image", None)
-            user_upload = civ.pop("user_upload", None)
-
-            update_archive_item_update_kwargs(
-                instance=instance,
-                interface=interface,
-                value=value,
-                image=image,
-                user_upload=user_upload,
-                upload_session=upload_session,
-                civ_pks_to_add=civ_pks_to_add,
-                upload_pks=upload_pks,
-            )
-
-        on_commit(
-            start_archive_item_update_tasks.signature(
-                kwargs={
-                    "archive_item_pk": instance.pk,
-                    "civ_pks_to_add": list(civ_pks_to_add),
-                    "upload_pks": upload_pks,
-                }
-            ).apply_async
-        )
-
+        values = validated_data.pop("values")
+        for value in values:
+            interface = value.get("interface", None)
+            user_upload = value.get("user_upload", None)
+            image = value.get("image", None)
+            value = value.get("value", None)
```

Comment:
Do we validate that only one of these is set?

Reply:
Yes, when serializing the civ data: https://github.com/comic/grand-challenge.org/blob/72153c085ed6cec5708b5288b7f5a77a257aa6ca/app/grandchallenge/components/serializers.py#L114-L123

Reply:
But then by doing `or` on the values, what happens when the values are falsey (0, "", intentional None etc)? In this construction you might end up with None when 0 was intended.

Reply:
Hm, you mean in the call to `create_civ`? I just tried this out, if `user_upload` and `image` are unset (i.e. None) and `value` is `""` or `0`, the or construction would return `""` or `0` respectively, just like we want. Did I misunderstand? 

Intentional `None` as value is a bit more tricky down the line, though. In `create_civ_for_value` we delete any currently existing civ if the new value is `None`. This enables deleting a value from a display set etc. So you wouldn't be able to add a value of None. I thought that that wouldn't ever be necessary, but maybe I'm wrong. Can you give an example of when we would want to do that?

Reply:
The JSON (Anything) type allows nulls.

Reply:
I see. In that case, we want to check whether None is a valid value for the interface and only delete the old CIV if None is not a valid value for the interface. Also noticed that `required` should be set to False for the interface form field for interfaces that allow None (or for boolean interfaces) - fixed that as well.